sched/semaphore/sem_holder.c

```diff
@@ -145,19 +141,13 @@ nxsem_findholder(FAR sem_t *sem, FAR struct tcb_s *htcb)
         }
     }
 #else
-  int i;
+  /* We have one hard-allocated holder structures in sem_t */
 
-  /* We have two hard-allocated holder structures in sem_t */
-
-  for (i = 0; i < 2; i++)
+  if (sem->holder.htcb == htcb)
     {
-      pholder = &sem->holder[i];
-      if (pholder->htcb == htcb)
-        {
-          /* Got it! */
+      /* Got it! */
 
-          return pholder;
-        }
+      return pholder;
```

Comment:
returning a stack garbage is not good

Reply:
Sorry for making a basic mistake, let me correct it.